src/FormattedMessage.js

```diff
@@ -0,0 +1,39 @@
+const format = (ctx, values) => ctx.parent.$formatMessage(ctx.props, values)
+const placeholder = (name) => `@\0@\0${name}\0@\0@`
+const placeholderRegex = /@\0@\0(.*?)\0@\0@/g
+
+export default {
+    functional: true,
+
+    props: {
+        id: {type: String, required: true},
+        defaultMessage: String,
+        values: Object,
```

Comment:
This is what I mean: https://vuejs.org/v2/guide/components.html#Compilation-Scope

By default when you use a component inside a parent component, all of the interpolated properties should be the parent properties, and not the child properties, so passing in the values into the child component as the values object to then have 'count' referenceable in what is still the parent scope goes against this.

However, this is also parallel to the React Intl component definitions, and I can't see a better way to do it without getting confused between the formatted message syntax and Vue interpolation.